File: java/gov/nasa/gsfc/gmsec/api5/ConfigFileIterator.java
/*
 * Copyright 2007-2023 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file ConfigFileIterator.java
 */

package gov.nasa.gsfc.gmsec.api5;

import gov.nasa.gsfc.gmsec.api5.ConfigFile.*;
import gov.nasa.gsfc.gmsec.api5.jni.JNIConfigFileIterator;


/**
 * This class is used to access elements of a configuration file that has been loaded
 * into a {@link ConfigFile} object.
 *
 * @see ConfigFile#getIterator()
 */
public class ConfigFileIterator
{
	private JNIConfigFileIterator m_jniConfigFileIterator = null;


	/** 
	 * This method is for internal GMSEC API use only.
	 * @param iter Object to reference for acquiring internal JNIConfigFileIterator
	 * @return Internal JNIConfigFileIterator object
	 */
	public static JNIConfigFileIterator getInternal(ConfigFileIterator iter)
	{
		return (iter == null ? null : iter.m_jniConfigFileIterator);
	}


	/**
	 * This method is for internal GMSEC API use only.
	 * @param jIter Internal JNIConfigFileIterator object
	 */
	public ConfigFileIterator(JNIConfigFileIterator jIter)
	{
		m_jniConfigFileIterator = jIter;
	}


	private ConfigFileIterator()
	{
	}


	/**
	 * Reports back whether there is a next config attribute in the ConfigFile.
	 *
	 * @return Returns true if another config is available; false otherwise.
	 */
	public boolean hasNextConfig()
	{
		return m_jniConfigFileIterator.hasNextConfig();
	}


	/**
	 * Reports back whether there is a next message attribute in the ConfigFile.
	 *
	 * @return Returns true if another message is available; false otherwise.
	 */
	public boolean hasNextMessage()
	{
		return m_jniConfigFileIterator.hasNextMessage();
	}


	/**
	 * Reports back whether there is a next subscription attribute in the ConfigFile.
	 *
	 * @return Returns true if another subscription is available; false otherwise.
	 */
	public boolean hasNextSubscription()
	{
		return m_jniConfigFileIterator.hasNextSubscription();
	}


	/**
	 * Reports back the next available ConfigFile.ConfigEntry in the ConfigFile.
	 *
	 * @return Returns the next ConfigFile.ConfigEntry.
	 *
	 * @throws GmsecException Thrown if no more ConfigFile.ConfigEntry objects are available.
	 *
	 * @see ConfigFileIterator#hasNextConfig()
	 */
	public ConfigFile.ConfigEntry nextConfig() throws GmsecException
	{
		return m_jniConfigFileIterator.nextConfig();
	}


	/**
	 * Reports back the next available ConfigFile.MessageEntry in the ConfigFile.
	 *
	 * @return Returns the next ConfigFile.MessageEntry.
	 *
	 * @throws GmsecException Thrown if no more ConfigFile.MessageEntry objects are available.
	 *
	 * @see ConfigFileIterator#hasNextMessage()
	 */
	public ConfigFile.MessageEntry nextMessage() throws GmsecException
	{
		return m_jniConfigFileIterator.nextMessage();
	}


	/**
	 * Reports the next available ConfigFile.SubscriptionEntry in the ConfigFile.
	 *
	 * @return Returns the next ConfigFile.SubscriptionEntry.
	 *
	 * @throws GmsecException Thrown if no more ConfigFile.SubscriptionEntry objects are available.
	 *
	 * @see ConfigFileIterator#hasNextSubscription()
	 */
	public ConfigFile.SubscriptionEntry nextSubscription() throws GmsecException
	{
		return m_jniConfigFileIterator.nextSubscription();
	}


	/**
	 * Resets all iterators to the beginning of the respective lists that are maintained
	 * by the ConfigFile.
	 */
	public void reset()
	{
		m_jniConfigFileIterator.reset();
	}
}

File: framework/src/gmsec5/ConfigFileIterator.cpp
/*
 * Copyright 2007-2023 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file ConfigFileIterator.cpp
 *
 *  @This file contains a definition for an iterator for the ConfigFile.
 */

#include <gmsec5/ConfigFileIterator.h>

#include <gmsec5/internal/InternalConfigFileIterator.h>

#include <gmsec5/Config.h>
#include <gmsec5/Message.h>


using namespace gmsec::api5;
using namespace gmsec::api5::internal;


ConfigFileIterator::ConfigFileIterator(internal::InternalConfigFile& cfgFile)
	: m_iter(new InternalConfigFileIterator(cfgFile))
{
}


ConfigFileIterator::~ConfigFileIterator()
{
	delete m_iter;
}


bool ConfigFileIterator::hasNextConfig() const
{
	return m_iter->hasNextConfig();
}


bool ConfigFileIterator::hasNextMessage() const
{
	return m_iter->hasNextMessage();
}


bool ConfigFileIterator::hasNextSubscription() const
{
	return m_iter->hasNextSubscription();
}


const ConfigFile::ConfigEntry& ConfigFileIterator::nextConfig() const
{
	return m_iter->nextConfig();
}


const ConfigFile::MessageEntry& ConfigFileIterator::nextMessage() const
{
	return m_iter->nextMessage();
}


const ConfigFile::SubscriptionEntry& ConfigFileIterator::nextSubscription() const
{
	return m_iter->nextSubscription();
}


void ConfigFileIterator::reset()
{
	m_iter->reset();
}
